SimG4CMS/PPS/src/PPSDiamondSD.cc

```diff
@@ -0,0 +1,304 @@
+///////////////////////////////////////////////////////////////////////////////
+//Author: Seyed Mohsen Etesami
+// setesami@cern.ch
+// 2016 Nov
+///////////////////////////////////////////////////////////////////////////////
+#include "SimG4CMS/PPS/interface/PPSDiamondSD.h"
+#include "FWCore/Framework/interface/EventSetup.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+#include "SimG4Core/Notification/interface/TrackInformation.h"
+#include "SimG4Core/Geometry/interface/SensitiveDetectorCatalog.h"
+#include "SimG4Core/Physics/interface/G4ProcessTypeEnumerator.h"
+#include "SimG4Core/SensitiveDetector/interface/SensitiveTkDetector.h"
+#include "SimDataFormats/TrackingHit/interface/UpdatablePSimHit.h"
+#include "SimDataFormats/SimHitMaker/interface/TrackingSlaveSD.h"
+#include "G4SDManager.hh"
+#include "G4Step.hh"
+#include "G4Track.hh"
+#include "G4VProcess.hh"
+#include "G4ParticleDefinition.hh"
+#include "G4ParticleTypes.hh"
+#include "SimG4Core/Notification/interface/TrackInformation.h"
+#include "G4PhysicalConstants.hh"
+#include "G4SystemOfUnits.hh"
+#include <iostream>
+#include <vector>
+#include <string>
+
+PPSDiamondSD::PPSDiamondSD(const std::string& name,
+                           const edm::EventSetup& es,
+                           const SensitiveDetectorCatalog& clg,
+                           edm::ParameterSet const& p,
+                           const SimTrackManager* manager)
+    : SensitiveTkDetector(name, es, clg, p),
+      numberingScheme(nullptr),
+      hcID(-1),
+      theHC(nullptr),
+      currentHit(nullptr),
+      theTrack(nullptr),
+      currentPV(nullptr),
+      unitID(0),
+      preStepPoint(nullptr),
+      postStepPoint(nullptr),
+      eventno(0) {
+  collectionName.insert(name);
+  edm::ParameterSet m_Anal = p.getParameter<edm::ParameterSet>("PPSDiamondSD");
+  verbosity_ = m_Anal.getParameter<int>("Verbosity");
+
+  LogDebug("PPSSimDiamond") << "*******************************************************\n"
+                            << "*                                                     *\n"
+                            << "* Constructing a PPSDiamondSD  with name " << name << "\n"
+                            << "*                                                     *\n"
+                            << "*******************************************************"
+                            << "\n";
+
+  slave = new TrackingSlaveSD(name);
+
+  if (name == "CTPPSTimingHits") {
+    numberingScheme = dynamic_cast<PPSVDetectorOrganization*>(new PPSDiamondNumberingScheme());
```

Comment:
should be a unique_ptr constructed with make_unique
also, the dynamic_cast is totally unnecessary

Reply:
Done.